test/preprocessing/test_preprocessing.py

```diff
@@ -42,6 +45,51 @@ def test_verify_pixel_variation(self):
         self.assertGreater(np.amax(self.raw), np.amin(self.raw))
         self.assertGreater(np.amax(self.comp), np.amin(self.comp))
 
+    def test_truncate_filenames(self):
+        """
+        Test that truncate_filenames does not crash
+
+        This test creates a directory structure with an image in it,
+        truncates filenames inside it, and basically checks that it
+        does not crash.
+
+        Ideally, this test would create a file with too long filename,
+        run truncate_filenames, and check that the filename now is
+        sufficiently short. However, it was hard to create a file wih
+        long filename, so I did not spend more time on this.
+        """
+
+        # Select a unique directory name for a new directory
+        outer_dir_name = 'temp_dir'
+        outer_dir_path = self.test_dir.joinpath(outer_dir_name)
+        while outer_dir_path.is_dir():
+            outer_dir_name = outer_dir_name + '_'
+            outer_dir_path = self.test_dir.joinpath(outer_dir_name)
+        outer_dir_path.mkdir(parents=True, exist_ok=True)
+        inner_dir_path = outer_dir_path.joinpath('inner_dir')
+        inner_dir_path.mkdir(parents=True, exist_ok=True)
```

Comment:
As `parents` is `True`, it will generate `outer_dir_path` if it does not already exist

Reply:
That is the intention. Can you explain the problem?

Reply:
No problem(?)